### ISSUE TITLE
Support scalar variables

### ISSUE THREAD

**Issue by angus-g** [meggart/NetCDF.jl#37]
It's possible to have a NetCDF file with variables of 0 dimensions (i.e. scalars). Currently, NetCDF.jl basically ignores their presence in a file (`LoadError: NetCDF file test.nc does not have a variable named foo`), and `nccreate` errors as well:

```
julia> nccreate("test.nc", "foo")
LoadError: MethodError: no method matching NetCDF.NcDim(::Void, ::Int64; atts=Dict{Any,Any}(Pair{Any,Any}("missval",-9999)), values=Float64[], unlimited=false)
```

**Comment by ikselven** [comment]
`ncinfo()` does already work with scalar variables and displays them.

What does not work yet is creating, writing and in some cases even reading scalar variables from a NetCDF file. Going through the functions mentioned in the documentation, these are affected:

High-Level API:
- `nccreate()`
- `ncwrite()`

Medium-Level API:
- `NetCDF.open(fil::AbstractString,v::AbstractString)`
- any variant of `NetCDF.readvar()`
- any variant of `NetCDF.putvar()`
- `NcVar()`


I've poked a bit around in the source code, trying to adapt it to support scalar variables. But I hit a problem with `NcVar`:
With `NcVar` being a Subtype of AbstractArray, a scalar would be treated as a zero-dimensional array. Read access _could_ be made possible by adjusting code at a few places, though I am unsure if it's even possible to write a value to such a variable. Also, it does not seem to be the right way to solve this.

After discussing it with a colleague, we came up with another approach, that might work: Implementing scalar variables as an extra type.

@meggart Which places in the code apart from the functions listed above would need to be adapted?

**Comment by meggart** [comment]
I have not come across a file that contains a scalar variable yet, I am still unsure if the 0-dimensional array approach would work better or a dedicated type. Basically both approaches are not so different in practise since it should not matter if you define special methods for NcVar{T,0} or for ScalarNcVar...

Do you have a quick example of a publicly accessible NetCDF file with a scalar variable or a short python/R script that generates one so that I can see what happens?

**Comment by ikselven** [comment]
My original use case is adding a variable for the CRS to a NetCDF file containing a number of Sentinel 1 scenes. According to [this example in the CF 1.7 convention](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#latitude-and-longitude-on-the-wgs-1984-datum) the variable should have no dimension.

For testing purposes, I created a very basic, sample NetCDF file with [the python script in this Gist](https://gist.github.com/ikselven/afa9307a6d3712fb0da6cd7f0bd0f38e), which I used for checking what works and what doesn't for the above comment. If you rather prefer actual real-world examples, you can find quite a few with a lot of scalar variables on the website of the NOAA:
<https://data.nodc.noaa.gov/sar-winds/radarsat2/2018/05/>

**Comment by meggart** [comment]
Great, thanks a lot for the link and the script. I think this can be solved quite easily by adding a few methods for NcVar{T,0}, can you try this branch and see if it works? https://github.com/JuliaGeo/NetCDF.jl/pull/58

**Comment by ikselven** [comment]
It's a bit unusual to work with 0-dimensional arrays instead of actual scalar values, but apart from that it is working for me.

Thank you very much : )

**Comment by meggart** [comment]
> It's a bit unusual to work with 0-dimensional arrays instead of actual scalar values,

I know. The question somehow comes down to if you prefer consistency or useability in special cases. I think it is useful to guarantee sure that `ncread` always returns an array. Returning scalars for variables without axes might break generic code quite easily, where the shapes of the variables are unknown and would lead to special-casing code like:
````julia
data = ncread(...)
firstvalue = isa(data,Array) ? data[1] : data
````

**Comment by meggart** [comment]
I think this got fixed by #58
